File: lab3/src/main/java/lab3/functionality/Health.java
package lab3.functionality;

public enum Health {
    HEALTHY("Полностью здоров"),
    SCRATCHED("Ушиблен"),
    INJURED("Ранен"),
    BLEEDING("Кровотечение"),
    DEATH_DOOR("При смерти"),
    DEAD("Мёртв");

    private final String name;

    Health(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public Health previous() {
        if (this == HEALTHY) return HEALTHY;
        return values()[this.ordinal() - 1];
    }

    public Health next() {
        if (this == DEAD) return DEAD;
        return values()[this.ordinal() + 1];
    }
}

File: lab4/src/main/kotlin/lab4/functionality/Health.kt
package lab4.functionality

enum class Health(val state: String) {
    HEALTHY("Полностью здоров"),
    SCRATCHED("Ушиблен"),
    INJURED("Ранен"),
    BLEEDING("Кровотечение"),
    DEATH_DOOR("При смерти"),
    DEAD("Мёртв");

    fun previous() = if (this == HEALTHY) HEALTHY else values()[ordinal - 1]

    fun next() = if (this == DEAD) DEAD else values()[ordinal + 1]
}